neo/rawio/spikegadgetsrawio.py

```diff
@@ -0,0 +1,247 @@
+        if self.selected_streams is not None:
+            if isinstance(self.selected_streams, str):
+                self.selected_streams = [self.selected_streams]
+            assert isinstance(self.selected_streams, list)
+            
+            keep = np.in1d(signal_streams['id'], self.selected_streams)
+            signal_streams = signal_streams[keep]
+
+            keep = np.in1d(signal_channels['stream_id'], self.selected_streams)
+            signal_channels = signal_channels[keep]
+        
+        # No events channels
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        # No spikes  channels
+        spike_channels = []
+        spike_channels = np.array(spike_channels, dtype=_spike_channel_dtype)
+
+        # fille into header dict
+        self.header = {}
+        self.header['nb_block'] = 1
+        self.header['nb_segment'] = [1]
+        self.header['signal_streams'] = signal_streams
+        self.header['signal_channels'] = signal_channels
+        self.header['spike_channels'] = spike_channels
+        self.header['event_channels'] = event_channels
+
+        self._generate_minimal_annotations()
+        # info from GlobalConfiguration in xml are copied to block and seg annotations
+        bl_ann = self.raw_annotations['blocks'][0]
+        seg_ann =  self.raw_annotations['blocks'][0]['segments'][0]
+        for ann in (bl_ann, seg_ann):
+            ann.update(gconf.attrib)
+
+    def _segment_t_start(self, block_index, seg_index):
+        return 0.
+
+    def _segment_t_stop(self, block_index, seg_index):
+        size = self._raw_memmap.shape[0]
+        t_stop = size / self._sampling_rate
+        return t_stop
+
+    def _get_signal_size(self, block_index, seg_index, stream_index):
+        size = self._raw_memmap.shape[0]
+        return size
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+        return 0.
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, stream_index, channel_indexes):
+        stream_id = self.header['signal_streams'][stream_index]['id']
+
+        raw_unit8 = self._raw_memmap[i_start:i_stop]
+        
+        num_chan = len(self._mask_channels_bytes[stream_id])
+        if channel_indexes is None:
+            # no loop : entire stream mask
+            stream_mask = self._mask_streams[stream_id]
+        else:
+            # acculate mask
+            if  isinstance(channel_indexes, slice):
+                chan_inds = np.arange(num_chan)[channel_indexes]
+            else:
+                chan_inds = channel_indexes
+            stream_mask = np.zeros(raw_unit8.shape[1], dtype='bool')
+            for chan_ind in chan_inds:
+                chan_mask = self._mask_channels_bytes[stream_id][chan_ind]
+                stream_mask |= chan_mask
+            # TODO : make a fix when "channel_indexes" arein wring order.
+
+        # this do a copy from memmap to memory
```

Comment:
Is there a way to avoid copying the data here? I thought this method should be fast and ideally not copy the data...

Reply:
No impossible.